### DISCUSSION TITLE
How to increase the VM's disk size?

### DISCUSSION THREAD

**Discussion by eratio08** [runfinch/finch#231] [Q&A] [answered]
How would one increase the VM's disk size to more than 50GiB using finch?

**Comment by davidhsingyuchen** [comment, marked as answer]
Currently finch does not support configuring the VM's disk size ([supported configurations](https://github.com/runfinch/finch#configuration)). Feel free to open a feature request as an issue. As a side note, the default disk size is 100GiB though.

**Reply by eratio08** [reply]
Thanks for the reply! 

I was (wrongly?) referring to https://github.com/runfinch/finch/blob/fb78051eaf1aae7b6eb99707f36489bb52818021/pkg/disk/disk.go#L25 by calling out the 50GiB

**Reply by davidhsingyuchen** [reply]
Ah I see, that makes sense! Just for the posterity, that size is for the persistent disk, which is a disk additional to the VM disk, and it's there to persist user data such as containers and images across Finch upgrades as VM disk will be erased during Finch upgrades.

**Reply by eratio08** [reply]
Ok then I guess this ran out of space for me and not the VM disk, due to huge image. I was not able to download/extract some image we use during development.

**Reply by davidhsingyuchen** [reply]
I see. Please feel free to open an issue to make the following configurable:

https://github.com/runfinch/finch/blob/fb78051eaf1aae7b6eb99707f36489bb52818021/pkg/disk/disk.go#L25

 Thanks!

**Reply by davidhsingyuchen** [reply]
Opened an issue accordingly: https://github.com/runfinch/finch/issues/275

Before the feature is implemented, you may follow the steps below to increase the size of the disk (thanks @pendo324 for helping out). `+100G` below means increasing the size by `100G` (i.e., if the original disk size is `50G`, it would be `150G` after running the commands below).

```sh
# stop VM
$ finch vm stop
# Resize the disk image
$ /Applications/Finch/lima/bin/qemu-img resize /Applications/Finch/lima/data/_disks/finch/datadisk +100G
# Start the VM
$ finch vm start
# Grow the partition
$ LIMA_HOME=/Applications/Finch/lima/data /Applications/Finch/lima/bin/limactl shell finch sudo bash -c "growpart /dev/vdb 1"
# Resize the fs
$ LIMA_HOME=/Applications/Finch/lima/data /Applications/Finch/lima/bin/limactl shell finch sudo bash -c "resize2fs /dev/vdb1"
```

After these steps, run the following command to check if the disk size is indeed increased:

```sh
LIMA_HOME=/Applications/Finch/lima/data /Applications/Finch/lima/bin/limactl shell finch df -H
```
